internal/app/osd/internal/reg/reg.go

```diff
@@ -292,32 +268,49 @@ func (r *Registrator) Dmesg(ctx context.Context, in *empty.Empty) (data *proto.D
 // Logs implements the proto.OSDServer interface. Service or container logs can
 // be requested and the contents of the log file are streamed in chunks.
 func (r *Registrator) Logs(req *proto.LogsRequest, l proto.OSD_LogsServer) (err error) {
-	ctx := namespaces.WithNamespace(context.Background(), req.Namespace)
-	client, err := containerd.New(defaults.DefaultAddress)
+	var containers []containerProc
+	var client *containerd.Client
+	var ctx context.Context
+
+	containers, err = containerID(req.Namespace)
 	if err != nil {
 		return err
 	}
-	// nolint: errcheck
-	defer client.Close()
-	task, err := client.TaskService().Get(ctx, &tasks.GetRequest{ContainerID: req.Id})
+	client, ctx, err = connect(req.Namespace)
 	if err != nil {
 		return err
 	}
-	file, _err := os.OpenFile(task.Process.Stdout, os.O_RDONLY, 0)
-	if _err != nil {
-		err = _err
-		return
-	}
-	chunk := filechunker.NewChunker(file)
+	// nolint: errcheck
+	defer client.Close()
 
-	if chunk == nil {
-		return errors.New("no log reader found")
-	}
+	for _, container := range containers {
+		if container.Name != req.Id {
+			continue
+		}
+
+		//var task runtime.Status
+		task, wtf := client.TaskService().Get(ctx, &tasks.GetRequest{ContainerID: req.Id})
```

Comment:
nit:

```
var task *IDONTKNOWTHETYPE
if task, err = client.TaskService().Get(ctx, &tasks.GetRequest{ContainerID: req.Id}); err != nil
...
```
or at the very least can we change `wtf` to `err`

Reply:
forgot about that.  came from frustration because I couldn't find out the type that silly thing returned... `*tasks.GetResponse`.

Reply:
:D I figured it was something like that.